src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -1354,7 +1354,7 @@ void UpdateMemoryCost(in UInt256 position, in UInt256 length)
                         UpdateMemoryCost(in dest, length);
                         ZeroPaddedSpan codeSlice = code.SliceWithZeroPadding(src, (int) length);
                         vmState.Memory.Save(in dest, codeSlice);
-                        if (_txTracer.IsTracingInstructions) _txTracer.ReportMemoryChange((long) dest, codeSlice.ToArray());
+                        if (_txTracer.IsTracingInstructions) _txTracer.ReportMemoryChange((long) dest, codeSlice);
```

Comment:
Code is cache / reused - if someone modifies span outside then it is dangerous.

Reply:
same as above, was thinking about going with ReadOnlySpan<>